internal/server/change_firm.go

```diff
@@ -4,49 +4,114 @@ import (
 	"fmt"
 	"net/http"
 	"strconv"
+	"strings"
 
 	"github.com/gorilla/mux"
 	"github.com/ministryofjustice/opg-sirius-supervision-pro-deputy-hub/internal/sirius"
 )
 
-func renderTemplateForChangeFirm(client ProDeputyHubInformation, tmpl Template) Handler {
+type ProDeputyChangeFirmInformation interface {
+	GetProDeputyDetails(sirius.Context, int) (sirius.ProDeputyDetails, error)
+	GetFirms(sirius.Context) ([]sirius.Firm, error)
+	AssignDeputyToFirm(sirius.Context, int, int) error
+}
+
+type changeFirmVars struct {
+	Path             string
+	XSRFToken        string
+	ProDeputyDetails sirius.ProDeputyDetails
+	FirmDetails      []sirius.Firm
+	Error            string
+	Errors           sirius.ValidationErrors
+	Success          bool
+	SuccessMessage   string
+	ExistingFirm     bool
+}
+
+func renderTemplateForChangeFirm(client ProDeputyChangeFirmInformation, tmpl Template) Handler {
 	return func(perm sirius.PermissionSet, w http.ResponseWriter, r *http.Request) error {
 
 		ctx := getContext(r)
 		routeVars := mux.Vars(r)
 		deputyId, _ := strconv.Atoi(routeVars["id"])
+		firm := checkUrlForFirm(r.URL.String())
```

Comment:
Not sure I understand what this is for. Why can't you find out whether there is an existing firm by looking in the deputy details?

Reply:
Yeah sorry this was to do with loading an existing firm, I've replaced it with javascript as Jen requested so I'll remove this bit